### ISSUE TITLE
Error formatting 3 files in trio

### ISSUE THREAD

**Issue by njsmith** [ambv/black#95]
Using an install of black master (fa159bb3576b0d8a79) running on Debian testing's python 3.6, trying to format the trio source code, it fails on 3 different files. Reproduction script:

```
$ git clone git@github.com:python-trio/trio.git
$ cd trio
$ git checkout 24cf95a62e9c6b683f7f875201dc2649a33230d8
$ black trio
```

Errors in the output:

```
error: cannot format trio/tests/test_socket.py: 18
error: cannot format trio/tests/test_highlevel_open_tcp_stream.py: INTERNAL ERROR: Black produced invalid code: invalid syntax (<unknown>, line 298). Please report a bug on https://github.com/ambv/black/issues.  This invalid output might be helpful: /tmp/blk_usla4iw_.log
error: cannot format trio/_core/tests/test_run.py: INTERNAL ERROR: Black produced code that is not equivalent to the source.  Please report a bug on https://github.com/ambv/black/issues.  This diff might be helpful: /tmp/blk_8btywbcq.log
```

Here are the log files:

[blk_8btywbcq.log](https://github.com/ambv/black/files/1865051/blk_8btywbcq.log)
[blk_usla4iw_.log](https://github.com/ambv/black/files/1865052/blk_usla4iw_.log)

**Comment by ambv** [comment]
Thanks for reporting, I noticed this too, hence no release yet. There are three problems:

1. Trailing comma gets put in a wrong place if there's a trailing comment. Example:
```
    assert scenario.connect_times == {
        "1.1.1.1": 0,
        "2.2.2.2": 0.300,
        # 3.3.3.3 was never even started
        ,
    }
```
2. When `# yapf: disable` was found in trio/_core/tests/test_run.py, Black decided to discard that content entirely. 🤣

3. Or should I say, 18. This is again a `# yapf: disable`-related bug. This time it's when you disable twice in a row. Black doesn't support statement-level format disabling and it gets confused by this.
```
async def test_getaddrinfo(monkeygai):
    check(res, [
        (tsocket.AF_INET,  # 127.0.0.1 is ipv4
         tsocket.SOCK_STREAM,
         tsocket.IPPROTO_TCP,
         "",
         ("127.0.0.1", 12345)),
    ])  # yapf: disable

    check(res, [
        (tsocket.AF_INET6,
         tsocket.SOCK_DGRAM,
         tsocket.IPPROTO_UDP,
         "",
         ("::1", 12345, 0, 0)),
    ])  # yapf: disable
```

I'll fix all three this weekend.

**Comment by ambv** [comment]
Thanks for reporting, all this is fixed in 18.4a0!
